usim/_primitives/concurrent_exception.py

```diff
@@ -309,3 +317,25 @@ def __repr__(self):
         return \
             f'<object usim.{self.__class__.__name__} '\
             f'of {", ".join(map(repr, self.children))}>'
+
+    def flattened(self) -> 'Concurrent':
+        """
+        Collapse nested Concurrent exceptions
+
+        Recursively collapses nested ``Concurrent`` exceptions to provide a single
+        ``Concurrent`` exception containing all :py:attr:`~.children` of the hierarchy.
+        For example, flattening a ``Concurrent(Concurrent(KeyError()), IndexError())``
+        provides a ``Concurrent(KeyError(), IndexError())``.
+        """
+        if not any(isinstance(exc, Concurrent) for exc in self.children):
+            return self
+        leaves = []
```

Comment:
Please consider renaming. I think you mean leafs? Nothing that is actually leaving, correct?

Reply:
Oops, that was embarrassing. I've renamed it to ``leafs``.